source/COMRegistrationFixes/__init__.py

```diff
@@ -0,0 +1,68 @@
+import os
+import subprocess
+import winVersion
+from logHandler import log
+
+# Particular  64 bit / 32 bit system paths
+systemRoot=os.path.expandvars('%SYSTEMROOT%')
+system32=os.path.join(systemRoot,'system32')
+sysWow64=os.path.join(systemRoot,'syswow64')
+systemDrive=os.path.expandvars('%SYSTEMDRIVE%\\')
+programFiles=os.path.join(systemDrive,'program files')
+programFilesX86=os.path.join(systemDrive,'program files (x86)')
+
+def registerServer(fileName,wow64=False):
+	"""
+	Registers the COM proxy dll with the given file name
+	Using regsvr32.
+	@ param fileName: the path to the dll
+	@type fileName: str
+	@param wow64: If true then the 32 bit (wow64) version of regsvr32 will be used.
+	@type wow64: bool
+	"""
+	regsvr32=os.path.join(sysWow64 if wow64 else system32,'regsvr32.exe')
+	try:
+		subprocess.check_call([regsvr32,'/s',fileName])
+	except subprocess.CalledProcessError as e:
+		log.error("Error registering %s, %s"%(fileName,e))
+	else:
+		log.debug("Registered %s"%fileName)
+
+def applyRegistryPatch(fileName):
+	"""
+	Applies the registry patch with the given file name
+	using regedit.
+	@ param fileName: the path to the dll
+	@type fileName: str
+	"""
+	regedit=os.path.join(system32,'regedit.exe')
+	try:
+		subprocess.check_call([regedit,'/s',fileName])
+	except subprocess.CalledProcessError as e:
+		log.error("Error applying registry patch: %s, %s"%(fileName,e))
+	else:
+		log.debug("Applied registry patch: %s"%fileName)
+
+def fixCOMRegistrations():
+	"""
+	Registers most common COM proxies, in case they had accidentally been unregistered or overwritten by 3rd party software installs/uninstalls.
+	"""
+	is64bit=os.environ.get("PROCESSOR_ARCHITEW6432","").endswith('64')
+	OSMajorMinor=winVersion.winVersion[:2]
+	log.debug("Fixing COM registration for Windows %s.%s, %s"%(OSMajorMinor[0],OSMajorMinor[1],"64 bit" if is64bit else "32 bit"))  
+	# Commands taken from NVDA issue #2807 comment https://github.com/nvaccess/nvda/issues/2807#issuecomment-320149243
+	# OLEACC (MSAA) proxies

```

Comment:
I assume this implies registering oleacc.dll doesn't have the desired result?

Reply:
Correct. See @jcsteh's comment in #2807 on this: https://github.com/nvaccess/nvda/issues/2807#issuecomment-319879231